test/blackbox-tests/test-cases/pkg/lockfile-generation.t/run.t

```diff
@@ -61,7 +61,7 @@ Print the contents of each file in the lockdir:
   
   (version 0.0.1)
   
-  (deps baz bar)
+  (deps bar baz)
```

Comment:
How come these are flipped now? I am a bit worried that if this is due to using a map, we might get non-reproducible lockfiles due to using an unordered map. Maybe it would make sense to sort them just to be sure.

Reply:
Indeed they are flipped due to representing dependencies with a map.

> I am a bit worried that if this is due to using a map, we might get non-reproducible lockfiles due to using an unordered map. Maybe it would make sense to sort them just to be sure.

Non-reproducible how? All lockfiles generated by dune will have their dependencies in order (following this PR) due to using a map.

Reply:
Note that we should preserve the order in which the dependencies appear in the opam file. Packages can define the same binaries (unlike in opam) and the order would decide the shadowing rules.

Reply:
Although thinking more about this I think it's a bit dangerous to encode meaning in the order of dependencies. I know I have a habit of sorting dependencies alphabetically in other build systems and it would be a nasty surprise for users if doing so had an effect on the build. Is this behaviour specific to lockfiles or is it established already in other parts of dune?

Reply:
It's inherited form OCaml. For example, the linking order of libraries determines the order of evaluation of module effects. It's certainly not ideal, so that's why we don't re-order things unless it's really needed.